### PR TITLE
Updated icon page

### PR DESCRIPTION
**Pull request by hellatan**
- changed some npm scripts
- added icon names
- added icon search (only works in react icons section)
- added hot module replacement and webpack dev server for dev purposes
- css modules (local css)

### CODE REVIEW

File: js/SvgWrapper.js [hunk 1]
```diff
@@ -9,8 +9,8 @@ const template = function ({className, icon, title, style}) {
 
 /**
  * A very simple wrapper for an SVG icon required from the static repository.
- * NOTE! In v2 the only way to use icons in back-bone ie. without react is to require using a different loader
- * (for webpack) - you *must also use the leaging "!" in your loader so the default SVG->React loader is not used.
+ * NOTE! In v2 the only way to use icons in backbone ie. without react is to require using a different loader
```
Comment (walfly) [inline js/SvgWrapper.js]: this isn't actually true, right? You can require the dist/templateString/*.js files without a loader as they are just plain js.

Comment (hellatan) [inline js/SvgWrapper.js]: Idk. I actually didn't read what it said as this was originally there. @nameofname would know more about this. I was only fixing typos 😀

Comment (nameofname) [inline js/SvgWrapper.js]: @walfly is correct.

Comment (hellatan) [inline js/SvgWrapper.js]: i'm leaving this in this PR. if we need to remove it, then it can be removed in a subsequent PR since this was just fixing typos.

**Comment by hellatan** [comment]
@nameofname @walfly other than the typo comments, anything else? I also tested this so that the index.html file will still work as a standalone page without needing webpack dev server running.

File: js/SvgStringToReact.jsx [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
-    }
-});
+const SvgStringToReact = props => {
+    const styles = {
+        fill: "currentcolor",
+        verticalAlign: "middle"
+    };
+    return (
+        <svg
+            viewBox="0 0 24 24"
```
Comment (nameofname) [inline js/SvgStringToReact.jsx]: looks like this will give all the icons the same viewBox. This shouldn't be true since they are all different.

File: js/SvgWrapper.js (hunk 1 above)
Comment (nameofname) [inline js/SvgWrapper.js]: @walfly is correct - can you just delete this comment.

File: js/SvgStringToReact.jsx (hunk 2 above)
Comment (hellatan) [inline js/SvgStringToReact.jsx]: this was originally there. i believe @walfly wrote this one. I only converted this to use `extends React.Component` instead of `React.createClass`. This should be updated in a different PR.

Comment (hellatan) [inline js/SvgStringToReact.jsx]: err, i made this into a stateless component and got rid of `React.createClass`. Everything else about it is from the original code.

File: js/SvgWrapper.js (hunk 1 above)
Comment (hellatan) [inline js/SvgWrapper.js]: ok - that i can do :)

File: js/SvgStringToReact.jsx (hunk 2 above)
Comment (walfly) [inline js/SvgStringToReact.jsx]: This isn't being used.  I added it because i was confused based on the documentation.  It should be removed.  I'll open a new pr that gets rid of it.

File: sample/js/BackboneIcons.js [hunk 3]
```diff
@@ -4,31 +4,33 @@
...
  * Renders each of the icons using the svg wrapper backbone view.
  */
 const BackboneIcons = Backbone.View.extend({
 
-    allIcons() {
-        return iconList.map((icon, idx) => {
+    _allIcons() {
+        return iconList.map(icon => {
             // NOTE* Here we have to use a leading "!" to tell webpack not to use the svg->react loader
             const Icon = require('!html!../../src/' + icon);
```
Comment (nameofname) [inline sample/js/BackboneIcons.js]: if you want you can get rid of this !html loader thing here and just use the strinigfied version in the dist folder instead.

**Comment by nameofname** [comment]
@hellatan this is cool. I'm looking forward to seeing what it looks like now. Once this merged in please also merge into the gh-pages branch and push to upstream which will update this page : beautifulcode.1stdibs.com/dibs-vg/

File: js/SvgStringToReact.jsx (hunk 2 above)
Comment (walfly) [inline js/SvgStringToReact.jsx]: https://github.com/1stdibs/dibs-vg/pull/33

**Comment by nameofname** [comment]
It's merged, please publish @hellatan

**Comment by hellatan** [comment]
published:

> \+ dibs-vg@1.9.0

**Comment by hellatan** [comment]
@nameofname @walfly - gh-pages updated: http://beautifulcode.1stdibs.com/dibs-vg/
